forem/forem
# update ruby_parser and sexp_processor gems

## What type of PR is this? (check all applicable)


## Description

The installed version of RubyParser, required by the hairtrigger gem to read migration files, does not provide a ruby 3 parser and emits warnings.

Upgrade to 3.16 or later to get V30 parser support. 

## Related Tickets & Documents

## QA Instructions, Screenshots, Recordings

The easiest place to see the warnings had been bin/setup's database setup step.

```
== Preparing database ==
NOTE: RubyParser::V30 undefined, using RubyParser::V27.
NOTE: RubyParser::V30 undefined, using RubyParser::V27.
Dumping schema
NOTE: RubyParser::V30 undefined, using RubyParser::V27.
NOTE: RubyParser::V30 undefined, using RubyParser::V27.
NOTE: RubyParser::V30 undefined, using RubyParser::V27.
NOTE: RubyParser::V30 undefined, using RubyParser::V27.
```

This  does not emit warnings on this branch:

```
== Preparing database ==
Dumping schema
```



### UI accessibility concerns?

None

## Added/updated tests?

- [x] No, and this is why: gem update only

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

      [Admin Guide](https://admin.forem.com/), or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
      or in a [forem.dev](http://forem.dev) post
- [x] This change does not need to be communicated, and this is why not: cleanup of warnings only

## [optional] Are there any post deployment tasks we need to perform?

## [optional] What gif best describes this PR or how it makes you feel?

![alt_text](gif_link)
